platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py

```diff
@@ -122,36 +97,58 @@ def eeprom_parse(self):
             if attr == "sys_mfg_date":
                 value = datetime.datetime.strptime(value, '%m-%d-%y').strftime('%m/%d/%Y 00:00:00')
 
-            product = product_dict.get(value)
+            product = _product_dict.get(value)
             if product is not None:
                 value = product
             if len(eeprom_params) > 0:
                 eeprom_params += ","
             eeprom_params += "{0:s}={1:s}".format(elem[1], value)
 
         orig_stdout = sys.stdout
-
         sys.stdout = StringIO()
         try:
-            new_e = eeprom_tlvinfo.TlvInfoDecoder.set_eeprom(self, "", [eeprom_params])
+            eeprom_data = eeprom_tlvinfo.TlvInfoDecoder.set_eeprom(self, "", [eeprom_params])
         finally:
+            decode_output = sys.stdout.getvalue()
             sys.stdout = orig_stdout
 
-        eeprom_base.EepromDecoder.write_eeprom(self, new_e)
+        eeprom_base.EepromDecoder.write_eeprom(self, eeprom_data)
+        self.__eeprom_tlv_dict = self.__parse_output(decode_output)
 
-        return True
+    def __parse_output(self, decode_output):
+        EEPROM_DECODE_HEADLINES = 6
+        lines = decode_output.replace('\0', '').split('\n')
+        lines = lines[EEPROM_DECODE_HEADLINES:]
+        res = dict()
 
-    def serial_str(self):
-        return self.eeprom.prod_ser_num
+        for line in lines:
+            try:
+                match = re.search('(0x[0-9a-fA-F]{2})([\s]+[\S]+[\s]+)([\S]+[\s]*[\S]*)', line)
```

Comment:
please update with the description what do we expect to match here

Reply:
done